### PR TITLE
Revenue tracking: Add currency field to goal creation

### PR DESCRIPTION
**Pull request by vinibrsl**
This commit adds a currency field to the goals form. Goals that have a currency set are now revenue goals, and are cached with sites to later be used during ingestion. This is hidden under a feature flag for now.

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -320,6 +320,9 @@ config :plausible, Plausible.ImportDeletionRepo,
   transport_opts: ch_transport_opts,
   pool_size: 1
 
+config :ex_money,
+  open_exchange_rates_app_id: get_var_from_path_or_env(config_dir, "OPEN_EXCHANGE_RATES_APP_ID")
```
Comment (vinibrsl) [inline config/runtime.exs:324]: The application should work normally without this environment variable, but currency conversion won't work in runtime. Not used now, but I'll make sure to set this before deploying.

File: lib/plausible/site/cache.ex
Comment (vinibrsl) [inline lib/plausible/site/cache.ex:1]: I'd appreciate your feedback on this file, @aerosol. I added the `sites.revenue_goals` association to the sites cache.

Comment (vinibrsl) [inline lib/plausible/site/cache.ex:1]: I'd appreciate your feedback on this file, @aerosol. I added the sites.revenue_goals association to the sites cache.

File: priv/repo/migrations/20230417092745_add_monetary_value_to_goals.exs
Comment (vinibrsl) [inline priv/repo/migrations/20230417092745_add_monetary_value_to_goals.exs:1]: Once approved, I'll extract this migration to another pull request so it's merged and deployed first.

File: priv/repo/migrations/20230417092745_add_monetary_value_to_goals.exs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+defmodule Plausible.Repo.Migrations.AddMonetaryValueToGoals do
+  use Ecto.Migration
+
+  def change do
+    alter table(:goals) do
+      add :currency, :string
```
Comment (ukutaht) [inline priv/repo/migrations/20230417092745_add_monetary_value_to_goals.exs:6]: Does the currency code have a fixed length we could use in this field?

**Review by ukutaht — APPROVED** [review]

File: priv/repo/migrations/20230417092745_add_monetary_value_to_goals.exs (hunk 2 above)
Comment (vinibrsl) [reply priv/repo/migrations/20230417092745_add_monetary_value_to_goals.exs]: Yes! Good call. Can we please continue in this pull request? https://github.com/plausible/analytics/pull/2955/files

**Pull request merged**